### PR TITLE
Libpath update to PD

### PR DESCRIPTION
**Pull request by nturaga**
Allow different versions of Bioconductor to use different directory locations for installation and persistence in GCE.

See corresponding Leo PR for more details: https://github.com/DataBiosphere/leonardo/pull/1821

Screenshots of the Leo + anvil-docker changes working together:
Screen Shot 2021-01-19 at 3 37 25 PM

Screen Shot 2021-01-19 at 3 43 25 PM

tested persistence between runtimes as well

### CODE REVIEW

File: anvil-rstudio-bioconductor/Dockerfile
Comment (rtitle) [inline anvil-rstudio-bioconductor/Dockerfile:23]: Are `TERRA_R_PLATFORM` and `TERRA_R_PLATFORM_BINARY_VERSION` evs not needed anymore?

File: anvil-rstudio-bioconductor/scripts/set_up_package_dir.sh
Comment (rtitle) [inline anvil-rstudio-bioconductor/scripts/set_up_package_dir.sh:9]: Could this be R_PATH=`printenv R_LIBS_USER | sed 's/~\///g'\` ?

E.g. prob don't need to use R if just printing an ev

File: anvil-rstudio-bioconductor/scripts/set_up_package_dir.sh
Comment (rtitle) [inline anvil-rstudio-bioconductor/scripts/set_up_package_dir.sh]: Does this script get run as `rstudio-user`? Probably should since it's creating directories under `/home/rstudio`

File: anvil-rstudio-bioconductor/scripts/set_up_package_dir.sh
Comment (rtitle) [inline anvil-rstudio-bioconductor/scripts/set_up_package_dir.sh:15]: Related to above: does the `rstudio-user` user have permission to do this?

**Comment by rtitle** [comment]
A few comments above but overall the approach looks good.. once this is merged we'll need to push to GCR and also update the static images in terra-docker, and the Leo custom image script.

File: anvil-rstudio-bioconductor/Dockerfile
Comment (gpcarr) [reply anvil-rstudio-bioconductor/Dockerfile:23]: don't believe so, @nturaga can confirm

File: anvil-rstudio-bioconductor/scripts/set_up_package_dir.sh
Comment (gpcarr) [reply anvil-rstudio-bioconductor/scripts/set_up_package_dir.sh:9]: based on my understanding, `R_LIBS_USER` is an environment variable set by R so that's why we used `R CMD` where as `BIOCONDUCTOR_DOCKER_VERSION` is not so we don't use `R CMD`

Comment (nturaga) [reply anvil-rstudio-bioconductor/scripts/set_up_package_dir.sh:9]: This isn't a publicly available env variable, only accessible through R. 

For example,

```
$ docker run -it --rm us.gcr.io/anvil-gcr-public/anvil-rstudio-bioconductor:0.0.9 printenv
PATH=/usr/lib/rstudio-server/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=6a5df70fe63b
TERM=xterm
R_VERSION=4.0.3
LC_ALL=en_US.UTF-8
LANG=en_US.UTF-8
R_HOME=/usr/local/lib/R
CRAN=https://packagemanager.rstudio.com/all/__linux__/focal/latest
TZ=Etc/UTC
S6_VERSION=v1.21.7.0
RSTUDIO_VERSION=latest
DEBIAN_FRONTEND=noninteractive
BIOCONDUCTOR_DOCKER_VERSION=3.12.30
RSTUDIO_PORT=8001
RSTUDIO_HOME=/etc/rstudio
TERRA_R_PLATFORM=anvil-rstudio-bioconductor
TERRA_R_PLATFORM_BINARY_VERSION=0.99.1
RSTUDIO_USERSETTING=/home/rstudio/.rstudio/monitored/user-settings/user-settings
HOME=/root
```

File: anvil-rstudio-bioconductor/scripts/set_up_package_dir.sh
Comment (gpcarr) [reply anvil-rstudio-bioconductor/scripts/set_up_package_dir.sh]: I believe it's run as root because line 16 works just fine. maybe to the command that create dirs under `/home/rstudio` i could add `-u rstudio`

Comment (nturaga) [reply anvil-rstudio-bioconductor/scripts/set_up_package_dir.sh]: Not sure about this, but there could be installation issues if `rstudio` user doesn't have access to write to the folder `/home/rstudio/R/....4.0-3.12/`. @gpcarr Could you test this?

File: anvil-rstudio-bioconductor/scripts/set_up_package_dir.sh
Comment (rtitle) [reply anvil-rstudio-bioconductor/scripts/set_up_package_dir.sh:9]: oh ok 👍

File: anvil-rstudio-bioconductor/scripts/set_up_package_dir.sh
Comment (gpcarr) [reply anvil-rstudio-bioconductor/scripts/set_up_package_dir.sh]: was able to confirm that this script runs as `root` user. I did test that `rstudio` user has access to write to `/home/rstudio/R/....4.0-3.12/`, there's a screenshot in the description showing packages successfully installed in this directory @nturaga. I'll do some refactoring so that the lines that write to `/home/rstudio` are run as `rstudio` user explicitly but by having the `chmod a+rwx` we are allowing this dir to be written to

File: anvil-rstudio-bioconductor/Dockerfile
Comment (nturaga) [reply anvil-rstudio-bioconductor/Dockerfile:23]: No need for this anymore because we inherit everything from bioconductor/bioconductor_docker images. These images have env variables we can use to find version identifiers and platform and other binaries related information.

File: anvil-rstudio-bioconductor/scripts/set_up_package_dir.sh
Comment (nturaga) [reply anvil-rstudio-bioconductor/scripts/set_up_package_dir.sh:15]: It probably doesn't, but it's not `rstudio-user` here is it? @gpcarr Can confirm ?

**Review by rtitle — APPROVED** [review]
Looks good!

**Comment by nturaga** [comment]
Looks good to me too!

**Comment by gpcarr** [comment]
> Looks good to me too!

awesome! i'll go ahead and merge and build the image and push it to GCR as `us.gcr.io/anvil-gcr-public/anvil-rstudio-bioconductor:0.0.10`. @rtitle we'll have to wait for the next Leo release to set up a terra-docker PR to bump this static image so I'll create a subtask to do that as a reminder

Edit: actually I should be able to PR to terra-docker once this image is pushed right so that the changes to the Leo custom image go out with the next Leo release as well?

**Comment by rtitle** [comment]
Yup, should be fine to PR and merge terra-docker. Can also regenerate the custom image and PR and merge to Leo. Then next Leo deploy we can update the images in the UI.

**Pull request merged**
